src/greedy.jl

```diff
@@ -169,76 +209,81 @@ function greedy(instance::Instance)
         end
 
         # Compute the set of indexes causing minimal-violation
-        candidates = [V[1]]
-        nb_min = nb_new_violation[1]
-        for c in 2:len
-            if nb_new_violation[c] < nb_min
-                nb_min = nb_new_violation[c]
-                candidates = [V[c]]
-            elseif nb_new_violation[c] == nb_min
-                push!(candidates, V[c])
-            end
-        end
+        candidates = filter_on_min_criterion(V, nb_new_violation)
 
-        # If |candidates| =/= 1 : TIE BREAK
-        #
-        # We will pass by tie breaking criterion
-        # This criterion encourages a more homogeneous distribution of
-        # the required options among the cars already in the partial sequence
-        # and those still to be scheduled.
-        # The distribution is based on the following idea:
-        #
-        #   If the average number of cars demanding a given option in the
-        #   partial sequence p is lower than that in the whole set V of
-        #   vehicles, then these cars should be encouraged to enter in
-        #   the sequence.
-        #
+        # Two candidates or more - First tie break
+        # If the average number of cars demanding a given option in the
+        # partial sequence π is lower than that in the whole set V of vehicles,
+        # then these cars should be encouraged to enter in the sequence.
         if length(candidates) > 1
             # Compute the tie break criterion for each candidates
             tie_break = zeros(Int,length(candidates))
             for i in 1:length(candidates)
-                for j in 1:nbH
+                for j in 1:instance.nb_HPRC
                     cond1 = !instance.RC_flag[candidates[i],j]
                     cond2 = (rv[j]-rpi[j])/len > (rpi[j])/length_pi
                     tie_break[i] += Int(xor( cond1 , cond2 ))
                 end
             end
 
             # Compute the new candidate list
-            # TODO use popfirst and push to filter candidates instead of copying the table
-            tmp_candidates = copy(candidates)
-            candidates = [tmp_candidates[1]]
-            max_tie_break = tie_break[1]
-            for i in 2:length(tmp_candidates)
-              if tie_break[i] > max_tie_break
-                  candidates = [tmp_candidates[i]]
-                  max_tie_break = tie_break[i]
-              elseif tie_break[i] == max_tie_break
-                  push!(candidates, tmp_candidates[i])
-              end
+            candidates = filter_on_max_criterion(candidates, tie_break)
+        end
+
+        # Two candidates or more - Second tie break
+        # Cars that require options with higher utilization rates should enter.
+        if length(candidates) > 1
+            # Compute the utilization rate of each options
+            utilization_rate = Array{AbstractFloat,1}(UndefInitializer(),instance.nb_HPRC)
```

Comment:
Why `AbstractFloat` and not `Float64`?

Reply:
Just an habit chomming from other languages, I'll change that